torch/_dynamo/variables/builtin.py

```diff
@@ -331,8 +332,22 @@ def call_function(
                 unimplemented(f"partial tensor op: {self} {args} {kwargs}")
 
         # Handle cases like int(torch.seed())
-        if self.fn is int and isinstance(args[0], DynamicShapeVariable):
-            return args[0]
+        # Also handle sym_float to sym_int cases
+        if self.fn in (int, float) and isinstance(args[0], DynamicShapeVariable):
```

Comment:
This felt like the morally correct thing, because in our python decompositions in core we're already manually replacing `int()/float()` calls with `sym_int()` and `sym_float()` - this makes dynamo do the same thing to user code.